src/api/store/community.py

```diff
@@ -1,16 +1,206 @@
+    for location in location_list:
+      if geography_type == 'ZIPCODE':
+        if location[0].isdigit():
+          location = location.replace(" ","")
+
+          # looks like a zipcode.  Check which towns it corresponds to
+          zipcode = zipcodes.matching(location)
+          if len(zipcode)>0:
+            city = zipcode[0].get('city', None)  
+          else:
+            raise Exception('No zip code entry found for zip='+location)
+
+          loc, created = Location.objects.get_or_create(location_type='ZIP_CODE_ONLY', zipcode=location, city=city)
+          if created:
+            print("Zipcode "+location+" created for town "+city)
+          else:
+            print("Zipcode "+location+" found for town "+city)
+                        
+          self._check_geography_unique(community, geography_type, location)
+
+        else:
+          # assume this is a town, see we can find the zip codes associated with it
+          ss = location.split('-')
+          town = ss[0]
+          if len(ss)==2:
+            state = ss[1]
+          else:
+            state = 'MA'
+            
+          zips = zipcodes.filter_by(city=town, state=state, zip_code_type='STANDARD' )
+          print("Number of zipcodes = "+str(len(zips)))
+          if len(zips)>0:
+            for zip in zips:
+              print(zip)
+              zipcode = zip["zip_code"]
+              loc, created = Location.objects.get_or_create(location_type='ZIP_CODE_ONLY', zipcode=zipcode, city=town)
+              if created:
+                print("Zipcode "+zipcode+" created")
+              else:
+                print("Zipcode "+zipcode+" found")
+
+              self._check_geography_unique(community, geography_type, zipcode)
+
+          else:
+            print("No zipcodes found corresponding to town "+ town + ", " + state)
+            raise Exception("No zipcodes found corresponding to city "+ town + ", " + state)
+      elif geography_type == 'CITY':
+        # check that this city is found in the zipcodes list
+        ss = location.split('-')
+        city = ss[0]
+        if len(ss)==2:
+          state = ss[1]
+        else:
+          state = 'MA'
+
+        zips = zipcodes.filter_by(city=city, state=state, zip_code_type='STANDARD' )
+        print("Number of zipcodes = "+str(len(zips)))
+        if len(zips)>0:
+          loc, created = Location.objects.get_or_create(location_type='CITY_ONLY', city=city, state=state)
+          if created:
+            print("City "+city+" created")
+          else:
+            print("City "+city+" found")
+
+        else:
+          print("No zipcodes found corresponding to city "+ city + ", " + state)
+          raise Exception("No zipcodes found corresponding to city "+ city + ", " + state)
+
+        self._check_geography_unique(community, geography_type, city)
+      elif geography_type == 'COUNTY':
+       # check that this county is found in the zipcodes list
+        ss = location.split('-')
```

Comment:
tbh this function seems weirdly long.  could you break it into chunks?  for instance all the logic in this elif could be put under a function called `county....`

Reply:
This comment is about style, not substance.